compaction/compaction_manager.hh

```diff
@@ -178,7 +178,7 @@ private:
     maintenance_scheduling_group _maintenance_sg;
     size_t _available_memory;
 
-    using get_candidates_func = std::function<std::vector<sstables::shared_sstable>(const column_family&)>;
+    using get_candidates_func = std::function<future<std::vector<sstables::shared_sstable>>()>;
```

Comment:
Is a future really necessary? I see one getter using a thread by AFAU it is not needed either.

Reply:
> Is a future really necessary? I see one getter using a thread by AFAU it is not needed either.

The reason it is needed is that cleanup impl of this func can potentially stall, so we have yielding points there.